File: Data Structures & Algorithms/subsets/submission-2.java
class Solution {
    public List<List<Integer>> res;
    public List<List<Integer>> subsets(int[] nums) {
        res = new ArrayList<>();
        backtrack(new ArrayList<>(), nums, 0);
        return res;
    }

    public void backtrack(List<Integer> curr, int[] nums, int index) {
        if (index > nums.length) {
            return;
        }

        res.add(new ArrayList<>(curr));
        for (int i = index; i < nums.length; i++) {
            curr.add(nums[i]);
            backtrack(curr, nums, i + 1);
            curr.remove(curr.size() - 1);
        }
        // [1,2,3]
        // [1]
        // [[],[1],[1,2],[1,2,3],]
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        L = 0
        

        for R in range(len(nums)):
            if nums[L] != nums[R]:
                L += 1
                nums[L] = nums[R]
        
        return L + 1